lib/models/builder.js

```diff
@@ -208,33 +227,41 @@ module.exports = CoreObject.extend({
     });
   },
 
+  _reportVizInfo: function (vizInfo) {
+    if (this.project && this.project.addons.length) {
+      this.project.addons.forEach(function (addon) {
+        if (addon._buildAnalytics) {
+          addon._buildAnalytics(vizInfo);
+        }
+      });
+    }
+  },
+
   /**
    * @private
    * @method build
    * @return {Promise}
    */
-  build: function() {
+  build: function(willReadStringDir, resultAnnotation) {
     var self = this;
-    var args = [];
-    for (var i = 0, l = arguments.length; i < l; i++) {
-      args.push(arguments[i]);
-    }
-
     attemptNeverIndex('tmp');
 
     return this.processAddonBuildSteps('preBuild')
       .then(function() {
-        return self.builder.build.apply(self.builder, args);
+        return self.builder.build(willReadStringDir);
       })
+      .then(this.processAddonBuildSteps.bind(this, 'postBuild'))
+      .then(this.processBuildResult.bind(this))
       .then(function(result) {
-
         if (vizEnabled()) {
-          outputViz(buildCount++, result);
+          var vizJson = computeVizInfo(buildCount, result, resultAnnotation);
+          self._reportVizInfo(vizJson);
+          writeVizInfo(buildCount, vizJson);
```

Comment:
It would be great if these 2 steps were controlled by different flags. If someone only needs to access the data programmatically, there is no reason to generate those files.